src/stp/new_plays/GoalPotentialPass.cpp

```diff
@@ -0,0 +1,158 @@
+        std::string defenderName = "defender" + std::to_string(defenderIndex + 1);
+
+        if (stpInfos.find(defenderName) != stpInfos.end()) {
+            stpInfos[defenderName].setPositionToMoveTo(defensivePositions[defenderIndex]);
+        }
+    }
+}
+
+std::vector<Vector2> GoalPotentialPass::calculateDefensivePositions(int numberOfDefenders, world_new::World* world, std::vector<world_new::view::RobotView> enemyRobots) {
+    std::vector<Vector2> positions = {};
+
+    // 3 robots will defend goal
+    for (int i = 0; i < numberOfDefenders; i++) {
+        if (i < 3) {
+            positions.push_back(world->getField()->getOurGoalCenter());
+        } else {
+            positions.push_back(enemyRobots[i].get()->getPos());
+        }
+    }
+
+    return positions;
+}
+
+bool GoalPotentialPass::shouldRoleSkipEndTactic() { return false; }
+
+const char* GoalPotentialPass::getName() { return "GoalPotentialPass"; }
+
+    Vector2 GoalPotentialPass::calculatePassLocation() {
+        auto ourBots = world->getWorld()->getUs();
+        auto theirBots = world->getWorld()->getThem();
+
+        double fieldWidth = field.getFieldWidth();
+        double offSetX = 0.3 * fieldWidth; // start looking for suitable positions to move to at 30% of the field width
+        double offSetY = -3;
+        double regionWidth = 3;
+        double regionHeight = 6;
+        auto numStepsX = 20;
+        auto numStepsY = 20;
+
+        double bestScore = 0;
+        Vector2 bestPosition{};
+
+        // Make a grid with all potentially good points
+        Grid grid = Grid(offSetX, offSetY, regionWidth, regionHeight, numStepsX, numStepsY);
+        for (const auto& nestedPoints : grid.getPoints()) {
+            for (const auto& trial : nestedPoints) {
+                auto w = world->getWorld().value();
+                auto visibility = FieldComputations::getPercentageOfGoalVisibleFromPoint(field, false, trial, w)/100;
+                auto fieldDiagonalLength = sqrt(pow(fieldWidth, 2.0) + pow(field.getFieldLength(), 2.0));
+
+                // Normalize distance, and then subtract 1
+                // This inverts the score, so if the distance is really large,
+                // the score for the distance will be close to 0
+                auto goalDistance = 1 - (FieldComputations::getDistanceToGoal(field, false, trial) / fieldDiagonalLength);
+
+                auto theirClosestBot = world->getWorld().value().getRobotClosestToPoint(trial, world_new::Team::them);
+                auto theirClosestBotDistance = (theirClosestBot->getPos().dist(trial) / fieldDiagonalLength);
+
+                // Make sure the angle to shoot at the goal with is okay
+                auto trialToGoal = field.getTheirGoalCenter() - trial;
+
+                // Some modifications to scores:
+                if (theirClosestBotDistance < 0.5) {
+                    goalDistance = 0;
+                }
```

Comment:
This is the sort of magic numer you probably want as a constant somewhere so it can be modified easily int he future.